pygmt/datatypes/dataset.py

```diff
@@ -226,8 +228,13 @@ def to_dataframe(
                 pd.Series(data=np.char.decode(textvector), dtype=pd.StringDtype())
             )
 
+        # Return an empty DataFrame if no columns are found.
+        if len(vectors) == 0:
+            return pd.DataFrame()
```

Comment:
Currently, it returns an empty DataFrame without columns and rows, but an empty DataFrame with columns is also allowed, e.g., 
```
return pd.DataFrame(column=column_names)
```
I guess either is fine. I think we can use `return pd.DataFrame()` now and make changes if necessary.

Reply:
I think we should return the column_names, so that users who want to e.g. do `pd.concat` on multiple `pd.DataFrame` outputs from running an algorithm like `pygmt.select` can do so in a more straightforward way. Note that we should also set the dtypes of the columns properly, even if the rows are empty, otherwise the dtypes will all become `object`:

```python
df1 = pd.DataFrame(data=[[0, 1, 2]], columns=["x", "y", "z"])
print(df1.dtypes)
# x    int64
# y    int64
# z    int64
# dtype: object
df2 = pd.DataFrame(columns=["x", "y", "z"])
print(df2.dtypes)
# x    object
# y    object
# z    object
# dtype: object

pd.concat(objs=[df1, df2]).dtypes
# x    object
# y    object
# z    object
# dtype: object
```

See my other suggestion on not returning an empty `pd.DataFrame()` early, until the dtype is set.